### PR TITLE
Feat/liwo 739 new filter options

### PR DESCRIPTION
**Pull request by vvandijck**
This pull requests enables users to select layer scenario variants using _multiple_ dropdowns, instead of just a single dropdown. This gives users more control over what kind of scenario variation they want to see.

Related ticket: [https://issuetracker.deltares.nl/browse/LIWO-739](https://issuetracker.deltares.nl/browse/LIWO-739)

TODO:
- [ ] Check changes after back-end fixes api data inconsistencies (lowercase vs uppercase etc.)

### CODE REVIEW

File: src/components/LayerControl.vue [hunk 1]
```diff
@@ -98,21 +122,54 @@ export default {
...
+      this.layer.variants.map(variant => (
+        Object.keys(variant.properties).find((key) => {
+          const keyIsAllowed = this.variantFilterProperties.includes(key)
+
+          if (!keyIsAllowed) { return false }
+
+          const existingOptions = uniqueOptions.find(field => field.title === key)
+
+          if (!existingOptions) {
+            uniqueOptions.push({ title: key, options: [variant.properties[key]] })
```
Comment (sjoerdbeentjes) [inline src/components/LayerControl.vue:142]: A side effect in an array method other than `forEach` something we should keep away from

File: src/lib/load-breach.js [hunk 2]
```diff
@@ -164,9 +166,25 @@ async function loadBreachLayer (breachId, layerName) {
...
+    method: 'GET',
+    mode: 'cors',
+    credentials: 'omit',
+    headers
+  })
+
+  const [breachData, filtersData] = await Promise.all([breachFetch, filterFetch])
+    .then(responses => responses.map(response => response.json()))
+
+  const properties = await filtersData
```
Comment (sjoerdbeentjes) [inline src/lib/load-breach.js:179]: is this `await` needed?

**Review by sjoerdbeentjes — CHANGES_REQUESTED** [review]

File: src/lib/load-breach.js (hunk 2 above)
Comment (vvandijck) [reply src/lib/load-breach.js:179]: Yes, otherwise it will return as `Promise {<pending>}`.

**Review by SiggyF — APPROVED** [review]
Looks good. 1 minor naming suggestion.

File: src/store.js [hunk 3]
```diff
@@ -30,7 +29,10 @@ export default new Vuex.Store({
     notificationsById: {},
 
     // This is the filter for probabilities (a string  used to pass to the backend)
-    probabilityFilter: ''
+    probabilityFilter: '',
+
+    // These are the variants used to filter the layer variant options
+    variantFilterProperties: {}
```
[the remark this answers is not in the document]
Comment (CindyvdVries) [reply src/store.js:35]: new data model not yet implemented in this branch

File: src/store.js [hunk 4]
```diff
@@ -99,6 +104,11 @@ export default new Vuex.Store({
 
   },
   getters: {
+    filterPropertiesIndex: (state) => (breachId) => {
```
[the remark this answers is not in the document]
Comment (CindyvdVries) [reply src/store.js:107]: variantFilterPropertiesIndex, will change

**Pull request merged**